uarray/_uarray_dispatch.cxx

```diff
@@ -232,35 +233,26 @@ py_func_args Function::replace_dispatchables(
 		return {std::move(res), nullptr};
 	}
 
-	auto itr = py_ref::steal(PyObject_GetIter(res));
-	if (itr == nullptr)
-	{
+	auto replaced_args = py_ref::steal(PySequence_Tuple(res));
+	if (!replaced_args)
 		return {};
-	}
-
-	const auto num_disp = PyTuple_Size(dispatchables);
-	auto replaced_args = py_ref::steal(PyTuple_New(num_disp));
-	Py_ssize_t pos = 0;
-	while (auto item = py_ref::steal(PyIter_Next(itr)))
-	{
-		if (pos > num_disp)
-		{
-			PyErr_SetString(PyExc_ValueError,
-			                "Too many values returned from __ua_convert__");
-			return {};
-		}
-		PyTuple_SET_ITEM(replaced_args.get(), pos, item.release());
-		++pos;
-	}
 
 	auto replacer_args = py_make_tuple(args, kwargs, replaced_args);
 	if (!replacer_args)
 		return {};
 
 	res = py_ref::steal(PyObject_Call(replacer_, replacer_args, nullptr));
-	if (PyTuple_Size(res) != 2)
+	if (!res)
+		return {};
+
+	if (!PyTuple_Check(res) || PyTuple_Size(res) != 2)
```

Comment:
You might want to use `Py_BuildValue` and `PyArg_ParseTuple` more. Although it might bring up the overhead, not sure... https://docs.python.org/3/c-api/arg.html